pkg/runtime/reconciler.go

```diff
@@ -84,12 +86,34 @@ func (r *reconciler) BindControllerManager(mgr ctrlrt.Manager) error {
 }
 
 // SecretValueFromReference fetches the value of a Secret given a
-// SecretReference
+// namespace, name and key of the secret
 func (r *reconciler) SecretValueFromReference(
-	ref *corev1.SecretReference,
```

Comment:
We should have an ACK core v1alpha1 type called `SecretKeyReference` that wraps the k8s `corev1.SecretReference` with an additional `Key` field.

Place this in a `github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1/secret.go` file:

```go
import (
    k8scorev1 "k8s.io/api/core/v1"
)

// SecretKeyReference combines a k8s corev1.SecretReference with a
// specific key within the referred-to Secret
type SecretKeyReference struct {
    k8scorev1.SecretReference
    // Key is the key within the secret
    Key string `json:"key"`
}
```